### PR TITLE
Improved Labels Rendering Logic

### PR DESCRIPTION
**Pull request by OssamaRafique**
### Optimization of `updateLabels` and Throttle Implementation

---

#### 📌 Changes:
- **SVG Dimension Retrieval**:
  - Refactored the `updateLabels` function. 
  - `svgWidth` and `svgHeight` are now obtained once via `getBoundingClientRect`, enhancing performance.
  
- **Throttle Integration**:
  - Added `throttleWithRAF` to optimize function calls, ensuring they align with the browser's frame rate.

---

### CODE REVIEW

File: src/epiviz.gl/utilities.js [hunk 1]
```diff
@@ -311,17 +311,33 @@ const getPointsBySelectMode = (selectMode, originalPoints, xRange, yRange) => {
   return points;
 };
 
+const throttleWithRAF = (callback) => {
```
Comment (jkanche) [inline src/epiviz.gl/utilities.js:314]: needs documentation for this function, also what does RAF mean?

**Review by jkanche — COMMENTED** [review]
mostly looks good, but may be a more understandable function name

File: src/epiviz.gl/utilities.js (hunk 1 above)
Comment (OssamaRafique) [reply src/epiviz.gl/utilities.js:341]: Updated & added documentation, RAF = request animation frame.

**Review by jkanche — APPROVED** [review]
lgtm

**Pull request merged**